### PR TITLE
(feat) O3-3140: Add `patientIdentifier` question type with custom validator 

### PR DESCRIPTION
**Pull request by gitcliff**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing.en-US#contributing-guidelines) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://om.rs/o3ui).
- [ ] My work includes tests or is validated by existing tests.

## Summary
This PR enables patientIdentifier to be added in the list of question types to be rendered when configuring questions 
## Screenshots
Screenshot 2024-05-08 at 1 53 36 AM

https://www.loom.com/share/b4dce4a2d2814e8598f593e5cb404b5f

## Related Issue
https://openmrs.atlassian.net/browse/O3-3140

### CODE REVIEW

File: src/components/interactive-builder/add-question-modal.component.tsx [hunk 1]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable no-console */
```
[the remark this answers is not in the document]
Comment (gitcliff) [reply src/components/interactive-builder/add-question-modal.component.tsx]: removed it

**Comment by denniskigen** [comment]
Do `patientIdentifer` fields require a backing concept? If so, we should show a concept lookup field when adding the question similar to [what's being done here](https://github.com/openmrs/openmrs-esm-form-builder/blob/main/src/components/interactive-builder/add-question-modal.component.tsx#L338).

**Comment by gitcliff** [comment]
> what's being done here



> Do `patientIdentifer` fields require a backing concept? If so, we should show a concept lookup field when adding the question similar to [what's being done here](https://github.com/openmrs/openmrs-esm-form-builder/blob/main/src/components/interactive-builder/add-question-modal.component.tsx#L338).



> Do `patientIdentifer` fields require a backing concept? If so, we should show a concept lookup field when adding the question similar to [what's being done here](https://github.com/openmrs/openmrs-esm-form-builder/blob/main/src/components/interactive-builder/add-question-modal.component.tsx#L338).

@denniskigen Not really, since we are not capturing them as obs , the crucial bit is the `PatientIdentifierType `

**Comment by gitcliff** [comment]
helo @denniskigen ,,i have updated the PR

**Comment by gitcliff** [comment]
@denniskigen kindly review

File: src/components/interactive-builder/edit-question-modal.component.tsx [hunk 2]
```diff
@@ -336,7 +366,64 @@ const EditQuestionModal: React.FC<EditQuestionModalProps> = ({
...
+                                key={index}
+                                onClick={() => handleIdentifierTypeSelect(identifierType)}
+                              >
+                                {identifierType?.display}
+                              </li>
+                            ))}
+                          </ul>
+                        );
+                      }
+                    })()}
```
Comment (denniskigen) [inline src/components/interactive-builder/edit-question-modal.component.tsx:420]:
```suggestion
                  <ComboBox
                    id="patientIdentifierTypeLookup"
                    items={patientIdentifierTypes}
                    itemToString={(item: PatientIdentifierType) => item?.display}
                    onChange={({ selectedItem }: { selectedItem: PatientIdentifierType }) => {
                      handleIdentifierTypeSelect(selectedItem);
                    }}
                    placeholder={t('choosePatientIdentifierType', 'Choose a patient identifier type')}
                    initialSelectedItem={patientIdentifierTypes.find(
                      (patientIdentifierType) =>
                        patientIdentifierType.uuid === questionToEdit.questionOptions.identifierType,
                    )}
```

Does something like this work better?

File: translations/en.json [hunk 3]
```diff
@@ -177,6 +180,7 @@
   "schemaSaveWarningMessage": "The dev3 server is ephemeral at best and can't be relied upon to save your schemas permanently. To avoid losing your work, please save your schemas to your local machine. Alternatively, upload your schema to the distro repo to have it persisted across server resets.",
   "searchConcept": "Search using a concept name or UUID",
   "searchForBackingConcept": "Search for a backing concept",
+  "searchForBackingPeatientIdentifierType": "Search for a backing patient identifier type",
```
Comment (denniskigen) [inline translations/en.json:183]:
```suggestion
  "searchForBackingPatientIdentifierType": "Search for a backing patient identifier type",
```

File: src/hooks/usePatientIdentifierName.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+  const { data, error } = useSWR<{ data: { name: { display: string } } }, Error>(
+    identifierType ? url : null,
+    openmrsFetch,
+  );
+
+  return {
+    patientidentifierName: data?.data?.name ?? null,
+    patientidentifierNameLookupError: error,
+    isLoadingpatientidentifierName: (!data && !error) || false,
```
Comment (denniskigen) [inline src/hooks/usePatientIdentifierName.ts:15]:
```suggestion
    patientIdentifierName: data?.data?.name ?? null,
    patientIdentifierNameLookupError: error,
    isLoadingPatientIdentifierName: isLoading,
```

File: src/hooks/usePatientIdentifierName.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import useSWR from 'swr';
+import { openmrsFetch, restBaseUrl } from '@openmrs/esm-framework';
+
+export function usePatientIdentifierName(identifierType: string | undefined) {
+  const url = `${restBaseUrl}/patientidentifiertype/${identifierType}`;
+
+  const { data, error } = useSWR<{ data: { name: { display: string } } }, Error>(
```
Comment (denniskigen) [inline src/hooks/usePatientIdentifierName.ts:7]:
```suggestion
  const { data, error, isLoading } = useSWR<{ data: { name: { display: string } } }, Error>(
```

**Comment by denniskigen** [comment]
Have you tested that the solution works as expected, @gitcliff?

**Comment by gitcliff** [comment]
> Have you tested that the solution works as expected, @gitcliff?

yes @denniskigen i have tested 
I will do a demo in the coffee break call

**Review by denniskigen — APPROVED** [review]
LGTM. Thanks, @gitcliff!

**Pull request merged**